dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnableTest.kt

```diff
@@ -121,60 +123,97 @@ internal class DataUploadRunnableTest {
         )
     }
 
-    @ParameterizedTest
-    @EnumSource(
-        SystemInfo.BatteryStatus::class,
-        names = ["DISCHARGING", "NOT_CHARGING"],
-        mode = EnumSource.Mode.INCLUDE
-    )
-    fun `doesn't send batch when battery is low and unplugged`(
-        batteryStatus: SystemInfo.BatteryStatus,
+    @Test
+    fun `M send batch W run() { batteryFullOrCharging }`(@Forgery batch: Batch) {
+        val fakeSystemInfo = SystemInfo(batteryFullOrCharging = true)
```

Comment:
Can we explicit the other values in system Info:
```
SystemInfo(
    batteryLevel = randomInt(0,100),
    batteryFullOrCharging = true, 
    onExternalPowerSource = randomBool(),
    powerSaveMode = false
)
```

Reply:
if you do that then you will actually add a false positive in the test, having the `onExternalPowerSource=true` will actually make the test positive event if the battery level condition fails.

Reply:
I will make them explicit though but without adding the randomness